nixos/lib/testing-python.nix

```diff
@@ -159,7 +161,12 @@ rec {
 
             echo -n "$testScript" > $out/test-script
             ${lib.optionalString (!skipLint) ''
-              ${python3Packages.black}/bin/black --check --diff $out/test-script
+              ( # Define the variables
+                echo "def testScript(subtest, start_all, ${lib.concatStringsSep ", " nodeNames}):"
+                sed -e 's/^/    /' -e 's/^    $//' <$out/test-script
+              ) >test-script
+              ${python3Packages.pyflakes}/bin/pyflakes test-script
```

Comment:
Can we move the actual linting into build that is detached from the rest? It is gaining a lot of complexity (comparatively).

Probably good to have a build for the script that passes if linting succeeded (unless disabled?).

Reply:
It's already possible to run only the linting

```
nix-build -A nixosTests.acme.driver
```

If that's all you want, it's done in this PR (or #122204 which is cherry-picked here).

If you want to have to attributes, one for running the test and one for linting, that's a new requirement that I would prefer to leave out of scope for now.